pkg/controllers/k8sdefaultendpoint/k8sdefaultendpoint.go

```diff
@@ -74,51 +80,155 @@ func syncKubernetesServiceEndpoints(ctx context.Context, virtualClient client.Cl
 		return err
 	}
 
-	// get virtual service endpoints
-	vObj := &corev1.Endpoints{}
-	err = virtualClient.Get(ctx, types.NamespacedName{
-		Namespace: "default",
-		Name:      "kubernetes",
-	}, vObj)
-	if err != nil {
-		if kerrors.IsNotFound(err) {
+	vEndpoints := &corev1.Endpoints{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "default",
+			Name:      "kubernetes",
+		},
+	}
+
+	result, err := controllerutil.CreateOrPatch(ctx, virtualClient, vEndpoints, func() error {
+		if vEndpoints.Labels == nil {
+			vEndpoints.Labels = map[string]string{}
+		}
+		vEndpoints.Labels[discovery.LabelSkipMirror] = "true"
+
+		// build new subsets
+		newSubsets := pEndpoints.DeepCopy().Subsets
+		for i := range newSubsets {
+			for j := range newSubsets[i].Ports {
+				newSubsets[i].Ports[j].Name = "https"
+			}
+			for j := range pEndpoints.Subsets[i].Addresses {
+				newSubsets[i].Addresses[j].Hostname = ""
+				newSubsets[i].Addresses[j].NodeName = nil
+				newSubsets[i].Addresses[j].TargetRef = nil
+			}
+			for j := range pEndpoints.Subsets[i].NotReadyAddresses {
+				newSubsets[i].NotReadyAddresses[j].Hostname = ""
+				newSubsets[i].NotReadyAddresses[j].NodeName = nil
+				newSubsets[i].NotReadyAddresses[j].TargetRef = nil
+			}
+		}
+
+		oldJSON, err := json.Marshal(vEndpoints.Subsets)
+		if err != nil {
+			return err
+		}
+		newJSON, err := json.Marshal(newSubsets)
+		if err != nil {
+			return err
+		}
+
+		if string(oldJSON) == string(newJSON) {
```

Comment:
Does this skip the patch?

Reply:
No. I didn't realize that when copying this piece of code :man_facepalming: 
The controllerutil.CreateOrPatch will skip the Patch call if there is no difference, so I'll just remove this^ part.